edge/pkg/metamanager/client/serviceaccount.go

```diff
@@ -49,7 +49,7 @@ func (c *serviceAccountToken) DeleteServiceAccountToken(podUID types.UID) {
 	for _, sa := range *svcAccounts {
 		var tr authenticationv1.TokenRequest
 		err = json.Unmarshal([]byte(sa.Value), &tr)
-		if err != nil {
+		if err != nil || tr.Spec.BoundObjectRef != nil {
```

Comment:
Is it not clear here to use one message describing that the error occurred and that 'BoundObjectRef' is nil? The `BoundObjectRef` is nil because it is nil, not because it is lost by unmarshalling.

Reply:
> Is it not clear here to use one message describing that the error occurred and that 'BoundObjectRef' is nil? The `BoundObjectRef` is nil because it is nil, not because it is lost by unmarshalling.

Sorry, I cannot get what you meant. I added this because during my local test, some dirty data appeared in the edge database. It had the serviceaccount record, but `boundObjectRef` was empty, causing a panic when reading the `tr.Spec.BoundObjectRef.UID` here.

Reply:
> Sorry, I cannot get what you meant. I added this because during my local test, some dirty data appeared in the edge database. It had the serviceaccount record, but `boundObjectRef` was empty, causing a panic when reading the `tr.Spec.BoundObjectRef.UID` here.

If this is a frequent occurrence, it can be processed with two pieces of logic and return the corresponding error message, if it is only due to a small amount of dirty data, it does not matter.
```golang
if err != nil {
    klog.Errorf("unmarshal resource %s token request failed: %v", sa.Key, err)
    continue
}else if tr.Spec.BoundObjectRef == nil {
    klog.Errorf("the specification BoundObjectRef of resource %s is nil", sa.Key)
    continue
}
```

Reply:
> > Sorry, I cannot get what you meant. I added this because during my local test, some dirty data appeared in the edge database. It had the serviceaccount record, but `boundObjectRef` was empty, causing a panic when reading the `tr.Spec.BoundObjectRef.UID` here.
> 
> If this is a frequent occurrence, it can be processed with two pieces of logic and return the corresponding error message, if it is only due to a small amount of dirty data, it does not matter.
> 
> ```go
> if err != nil {
>     klog.Errorf("unmarshal resource %s token request failed: %v", sa.Key, err)
>     continue
> }else if tr.Spec.BoundObjectRef == nil {
>     klog.Errorf("the specification BoundObjectRef of resource %s is nil", sa.Key)
>     continue
> }
> ```

Yes, it occurs only when there are some dirty data. I add this judgement since it will affect other normal data.